src/librbd/group/ListSnapshotsRequest.h

```diff
@@ -0,0 +1,93 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_LIBRBD_GROUP_LIST_SNAPSHOTS_REQUEST_H
+#define CEPH_LIBRBD_GROUP_LIST_SNAPSHOTS_REQUEST_H
+
+#include "include/int_types.h"
+#include "include/types.h"
+#include "include/rados/librados.hpp"
+#include "cls/rbd/cls_rbd_types.h"
+
+#include <string>
+#include <vector>
+
+class Context;
+
+namespace librbd {
+
+struct ImageCtx;
+
+namespace group {
+
+template <typename ImageCtxT = librbd::ImageCtx>
```

Comment:
ImageCtx is not used in this class.  And do we need this class to be templatized?

Reply:
I'm not sure. This class was taken from https://github.com/ceph/ceph/pull/53793/ and modified . I've seen other classes being templatized even though they don't use the ImageCtx (https://github.com/ceph/ceph/blob/main/src/librbd/api/Namespace.h) so I continued with it. I'm fine with removing it if required.